pandas/tests/io/formats/test_to_csv.py

```diff
@@ -541,15 +543,17 @@ def test_to_csv_compression_dict_no_method_raises(self):
     )
     def test_to_csv_zip_arguments(self, compression, archive_name):
         # GH 26023
-        from zipfile import ZipFile
-
         df = DataFrame({"ABC": [1]})
         with tm.ensure_clean("to_csv_archive_name.zip") as path:
             df.to_csv(
                 path, compression={"method": compression, "archive_name": archive_name}
             )
             with ZipFile(path) as zp:
-                expected_arcname = path if archive_name is None else archive_name
+                if archive_name is None:
```

Comment:
why this `if` and not simply replacing `path` by `Path(path).stem` in the original line?